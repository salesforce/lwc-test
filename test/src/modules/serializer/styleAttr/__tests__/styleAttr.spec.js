/*
 * Copyright (c) 2024, salesforce.com, inc.
 * All rights reserved.
 * SPDX-License-Identifier: MIT
 * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
 */
import { createElement } from 'lwc';
import StyleAttr from '../styleAttr';

// There is an expected warning message we can ignore
beforeEach(() => {
    jest.spyOn(console, 'error').mockImplementation((message) => {
        if (!message.includes("Invalid 'style' attribute passed to <div>")) {
            throw new Error('Unexpected console error message: ' + message);
        }
    });
});

afterEach(() => {
    jest.restoreAllMocks();
});

it('serializes component with different whitespace', () => {
    const elm = createElement('serializer-component', { is: StyleAttr });
    document.body.appendChild(elm);

    expect(elm).toMatchSnapshot();
});
