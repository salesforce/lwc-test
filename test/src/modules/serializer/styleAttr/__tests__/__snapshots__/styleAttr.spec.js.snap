// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`serializes component with different whitespace 1`] = `
<serializer-component>
  #shadow-root(open)
    <!-- static style, static optimized -->
    <div
      style="color: blue;"
    />
    <div
      style="color: blue; text-align: center;"
    />
    <div
      style="color: blue; text-align: center;"
    />
    <div
      style="color: blue; text-align: center;"
    />
    <div
      style="color: blue; text-align: center;"
    />
    <div
      style="color: blue; text-align: center;"
    />
    <div
      style="color: blue; text-align: center;"
    />
    <div
      style="color: blue; text-align: center;"
    />
    <div
      style="color: blue; text-align: center;"
    />
    <div
      style="background-color: red !important;"
    />
    <div
      style="background-color: red !important;"
    />
    <div
      style="background-color: red !important;"
    />
    <div
      style="background-color: red !important;"
    />
    <div
      style="background-color: red !important;"
    />
    <div
      style="color: yellow;"
    />
    <div
      style="color: purple;"
    />
    <div
      style="color: purple; text-align: right;"
    />
    <div />
    <div />
    <div />
    <div />
    <div
      style="background-image: url(data:image/svg+xml;base64,abc123); background-size: 12px;"
    />
    <!-- static style, not static optimized -->
    <div
      style="color: blue;"
    />
    <div
      style="color: blue; text-align: center;"
    />
    <div
      style="color: blue; text-align: center;"
    />
    <div
      style="color: blue; text-align: center;"
    />
    <div
      style="color: blue; text-align: center;"
    />
    <div
      style="color: blue; text-align: center;"
    />
    <div
      style="color: blue; text-align: center;"
    />
    <div
      style="color: blue; text-align: center;"
    />
    <div
      style="color: blue; text-align: center;"
    />
    <div
      style="background-color: red !important;"
    />
    <div
      style="background-color: red !important;"
    />
    <div
      style="background-color: red !important;"
    />
    <div
      style="background-color: red !important;"
    />
    <div
      style="background-color: red !important;"
    />
    <div />
    <div
      style="color: purple;"
    />
    <div
      style="color: purple; text-align: right;"
    />
    <div />
    <div />
    <div />
    <div />
    <div
      style="background-image: url(data:image/svg+xml;base64,abc123); background-size: 12px;"
    />
    <!-- dynamic style, static optimized -->
    <div />
    <div />
    <div />
    <div />
    <div />
    <div
      style="color: blue; text-align: center;"
    />
    <div
      style="color: blue; text-align: center;"
    />
    <div
      style="color: blue; text-align: center;"
    />
    <div
      style="color: blue; text-align: center;"
    />
    <div
      style="color: blue; text-align: center;"
    />
    <div
      style="color: blue; text-align: center;"
    />
    <div
      style="color: blue; text-align: center;"
    />
    <div
      style="color: blue; text-align: center;"
    />
    <div
      style="background-color: red !important;"
    />
    <div
      style="background-color: red !important;"
    />
    <div
      style="background-color: red !important;"
    />
    <div
      style="background-color: red !important;"
    />
    <div
      style="background-color: red !important;"
    />
    <div
      style="color: yellow;"
    />
    <div
      style="color: purple;"
    />
    <div
      style="color: purple; text-align: right;"
    />
    <div />
    <div />
    <div
      style="color: blue;"
    />
    <div
      style="background-image: url(data:image/svg+xml;base64,abc123); background-size: 12px;"
    />
    <!-- dynamic style, not static optimized -->
    <div />
    <div />
    <div />
    <div />
    <div />
    <div
      style="color: blue; text-align: center;"
    />
    <div
      style="color: blue; text-align: center;"
    />
    <div
      style="color: blue; text-align: center;"
    />
    <div
      style="color: blue; text-align: center;"
    />
    <div
      style="color: blue; text-align: center;"
    />
    <div
      style="color: blue; text-align: center;"
    />
    <div
      style="color: blue; text-align: center;"
    />
    <div
      style="color: blue; text-align: center;"
    />
    <div
      style="background-color: red !important;"
    />
    <div
      style="background-color: red !important;"
    />
    <div
      style="background-color: red !important;"
    />
    <div
      style="background-color: red !important;"
    />
    <div
      style="background-color: red !important;"
    />
    <div
      style="color: yellow;"
    />
    <div
      style="color: purple;"
    />
    <div
      style="color: purple; text-align: right;"
    />
    <div />
    <div />
    <div
      style="color: blue;"
    />
    <div
      style="background-image: url(data:image/svg+xml;base64,abc123); background-size: 12px;"
    />
</serializer-component>
`;
